jupyterhub_traefik_proxy/etcd.py

```diff
@@ -63,8 +63,12 @@ def _default_client(self):
         help="""the etcd key prefix for traefik configuration""",
     )
 
+    traefik_auth_api_port = Unicode(
```

Comment:
Let's make the config value a url: `traefik_api_url`. We don't want to assume localhost anywhere because we want to be sure that this works when traefik is on a remote host. The default should probably be:

    http://127.0.0.1:8099

We can use:

[`urlparse(self.traefik_api_url)`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) to extract components like `netloc` or `port` if we need them.